client/driver/qemu.go

```diff
@@ -381,27 +434,36 @@ func (h *qemuHandle) Signal(s os.Signal) error {
 	return fmt.Errorf("Qemu driver can't send signals")
 }
 
-// TODO: allow a 'shutdown_command' that can be executed over a ssh connection
-// to the VM
 func (h *qemuHandle) Kill() error {
-	if err := h.executor.ShutDown(); err != nil {
-		if h.pluginClient.Exited() {
-			return nil
+	// First, try sending a graceful shutdown command via the qemu monitor
+	err := sendQemuShutdown(h.logger, h.monitorPath, h.userPid)
```

Comment:
Wrap this in a if check seeing if graceful shutdown is enabled and log the error

Reply:
Done!